### ISSUE TITLE
Consider switching logging to slf4j

### ISSUE THREAD

**Issue by badgerwithagun** [alfasoftware/morf#9]
Currently we force a log4j dependency on applications. There's no need.

**Comment by badgerwithagun** [comment]
@venushka - would value your thoughts on this one.

**Comment by tsg21** [comment]
Yes, I think this is a sensible plan.

> Currently we force a log4j dependency on applications. There's no need.

We don't - commons.logging can be hooked into other logging implementations, but it's nowhere near as widely used as slf4j is.

**Comment by venushka** [comment]
I agree, we should switch to slf4j.

One thing to keep in mind though, we should make sure the test execution within Maven should be provided with a logging implementation (log4j like we already have probably) with the required configuration to ensure the logs from tests are redirected to a file and not be printed to console. Travis builds are limited to 4MB of console output for opensource projects, which is more than adequate for the build, but not including tests logs in console.

**Comment by badgerwithagun** [comment]
Yeah I saw that on your fork - makes sense.

**Comment by tsg21** [comment]
I've had a go at this in a fork. It's a reasonably easy find-replace.

**Comment by badgerwithagun** [comment]
@tsg21 - submit a PR!

**Comment by tsg21** [comment]
Don't see any reason why we'd want to stick to commons-logging?

**Comment by tsg21** [comment]
Anyone like to offer up a reason why we should keep the dependency on log4j?

**Comment by badgerwithagun** [comment]
Nope. Let's just do this. Could you create a PR with commons->slf4j and removal of log4j?

**Comment by tsg21** [comment]
Will do
